app/javascript/mastodon/components/follow_button.tsx

```diff
@@ -99,7 +99,12 @@ export const FollowButton: React.FC<{
   return (
     <Button
       onClick={handleClick}
-      disabled={relationship?.blocked_by || relationship?.blocking}
+      disabled={
+        relationship?.blocked_by ||
+        relationship?.blocking ||
+        account?.suspended ||
+        !!account?.moved
```

Comment:
Slightly worried about the moved bit, different implementations have different expectations about what a moved account means. Being completely unable to follow a moved account might be an issue.